### ISSUE TITLE
Add polygon type labels

### ISSUE THREAD

**Issue by nicOwlas** [labelflow/labelflow#251]
### Tasks
- [x] Create the polygon tool  ([mockup here](https://www.figma.com/file/4J9xyfF5v4B98MuKVO0Y8V/Labelflow-mockups?node-id=711%3A8707))
- [x] 1 click per point
- [x] Double click to save the polygon or click on the first point (default OpenLayers behaviour)
- [x] Display the stored polygon
- [x] Add to the DB to the Label object a field `type`: `polygon`
- [x] Allow update of polygon geometry like on Sterblue (you can modify any point on the contour - default openLayers - 
- [x] Activate snapping ([openlayer example](https://openlayers.org/en/latest/examples/draw-and-modify-features.html))
- [x] Echap during creation deletes every points and returns to the initial drawing state

### Create journal of events
- [x] During the creation process undo/redo is deactivated
- [x] After an object is created the undo/redo removes or adds back the object
- [x] Right click => does not trigger any action during the creation process of a polygon

### State Machine

⚠️ Read the [Finite State Machine document here](https://docs.google.com/spreadsheets/d/1siG5wMpmIPFSACUrJipTCUIVKyEtpBFw5lOcT-ZJIf4/edit#gid=0)
The lines specifically relative to the polygon tool are colored (ground truth is the google sheet, not the screenshot).

![image](https://user-images.githubusercontent.com/16015322/124137478-04723b00-da86-11eb-9ead-fd1160f24ea3.png)

### Not to be done in this issue
Delete a point during the creation of a polygon or when updating a polygon. It is not in the context of this issue but helpful links / functions to manage the undo/redo points per point when drawing a polygon (for future issues)
- [Openlayer example where points can be removed](https://openlayers.org/en/latest/examples/draw-features.html)
- ol:interactionDraw Method: removeLastPoint / appendCoordinates
- #260 
- #262
- #261

Question: how can we avoid invalid geometries?
Answer: will be done in another issue

blocked by #257
